sys/net/credman/credman.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_credman
+ * @{
+ *
+ * @file
+ * @brief       (D)TLS Credentials management module implementation
+ *
+ * @author  Aiman Ismail <muhammadaimanbin.ismail@haw-hamburg.de>
+ */
+
+#include "net/credman.h"
+
+#include <string.h>
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static credman_credential_t credentials[CREDMAN_MAX_CREDENTIALS];
+static unsigned used = 0;
+
+static int _find_credential_pos(credman_tag_t tag, credman_type_t type);
+static int _find_next_free_pos(void);
+
+int credman_add_credential(const credman_credential_t *credential)
+{
+    assert(credential);
+    int pos = -1;
+
+    if ((credential->type == CREDMAN_TYPE_EMPTY) ||
+        (credential->tag == CREDMAN_TAG_EMPTY)) {
+        DEBUG("credman: invalid credential type/tag\n");
+        return CREDMAN_ERROR;
+    }
+    switch (credential->type) {
+    case CREDMAN_TYPE_PSK:
+        if (credential->params.psk == NULL) {
+            DEBUG("credman: no PSK credential found\n");
+            return CREDMAN_ERROR;
+        }
+        break;
+    case CREDMAN_TYPE_ECDSA:
+        if (credential->params.ecdsa == NULL) {
+            DEBUG("credman: no ECDSA credential found\n");
+            return CREDMAN_ERROR;
+        }
+        if ((credential->params.ecdsa->private_key == NULL) ||
+            (credential->params.ecdsa->public_key.x == NULL) ||
+            (credential->params.ecdsa->public_key.y == NULL)) {
+            DEBUG("credman: invalid ECDSA parameters\n");
+            return CREDMAN_ERROR;
+        }
+        break;
+    default:
+        return CREDMAN_ERROR;
```

Comment:
Here you could return e.g. `CREDMAN_TYPE_UNKNOWN`.

Reply:
Fixed.